src/app/calculator/furnaces/wall/wall-form/wall-form.component.ts

```diff
@@ -0,0 +1,166 @@
+import { ChangeDetectorRef, Component, ElementRef, Input, OnInit, SimpleChanges, ViewChild } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+import { ModalDirective } from 'ngx-bootstrap';
+import { Subscription } from 'rxjs';
+import { WallLossesSurface } from '../../../../shared/models/materials';
+import { WallLoss } from '../../../../shared/models/phast/losses/wallLoss';
+import { Settings } from '../../../../shared/models/settings';
+import { SuiteDbService } from '../../../../suiteDb/suite-db.service';
+import { WallFormService } from '../wall-form.service';
+import { WallService } from '../wall.service';
+
+@Component({
+  selector: 'app-wall-form',
+  templateUrl: './wall-form.component.html',
+  styleUrls: ['./wall-form.component.css']
+})
+export class WallFormComponent implements OnInit {
+  
+  @Input()
+  settings: Settings;
+  @Input()
+  isBaseline: boolean;
+  @Input()
+  inModal: boolean;
+  @Input()
+  selected: boolean;
+  @ViewChild('surfaceModal', { static: false }) public surfaceModal: ModalDirective;
+  @ViewChild('flueGasModal', { static: false }) public flueGasModal: ModalDirective;
+  @ViewChild('formElement', { static: false }) formElement: ElementRef;
+
+  
+  surfaceOptions: Array<WallLossesSurface>;
+  showSurfaceModal: boolean = false;
+
+  wallLossesForm: FormGroup;
+  resetDataSub: Subscription;
+  generateExampleSub: Subscription;
+  showFlueGasModal: boolean;
+
+  constructor(private wallFormService: WallFormService,
+              private suiteDbService: SuiteDbService,
+              private wallService: WallService,
+              private cd: ChangeDetectorRef) { }
+  ngOnInit(): void {
+    this.initSubscriptions();
+    this.setFormState();
```

Comment:
setFormState() called in initForm() which is called in the initSubscriptions (twice). Don't think you need to call it here.